### PR TITLE
Multiple sources to one sink

### PR DESCRIPTION
**Pull request by dieris**
Construct the filter graph when multiple sources are connected to one sink (e.g. correlators)

### CODE REVIEW

**Comment by dieris** [comment]
It looks like this last commit is all we needed to fix variance. It was already meant to be connected to the same writer as the data 😅

**Comment by dieris** [comment]
Actually, doesn't it make more sense to save the variance of the real and imaginary parts?

File: src/auspex/filters/average.py [hunk 1]
```diff
@@ -136,6 +136,8 @@ def update_descriptors(self):
 
         # Define variance axis descriptor
         descriptor_var = descriptor_in.copy()
+        if descriptor_var.dtype == np.complex128:
+            descriptor_var.dtype = np.float64
```
Comment (grahamrow) [inline src/auspex/filters/average.py]: Does this conflict with any changes in #176 ?

Comment (dieris) [reply src/auspex/filters/average.py]: That's right, reverted last commit

**Pull request merged**